dockstore-client/src/main/java/io/dockstore/common/FileProvisionUtil.java

```diff
@@ -69,27 +75,48 @@ private FileProvisionUtil() {
     }
 
     static boolean downloadFromVFS2(String path, Path targetFilePath, int threads) {
+
+        // An extension of the Http4sFileProvider class to allow for setting the cookie specification to something
+        // other than DEFAULT. This cannot be set through the fileSystemOptions argument.
+        class CustomHttp4sFileProvider extends Http4sFileProvider {
```

Comment:
This should be `static private class...`.

1. private because it's only used inside this class, so as best practice don't make it more accessible that it need to be.
2. static is also a best practice for inner classes, unless there's an actual need for the class to access something from the enclosing class.

Reply:
Okay. Had to move the class outside the method to make this change- was getting "modifier not allowed here" error

Reply:
Whoops, I think it was OK the way it was. A lot  of  my Java knowledge is from Java 5, when local classes didn't exist, and the rules I stated above apply to inner classes. I hadn't paid close attention that this was a local class and didn't even know that local classes existed. :( 

So, sorry, I think you can revert it.

Reply:
No worries :)